src/include/sof/lock.h

```diff
@@ -118,30 +120,36 @@ extern uint32_t lock_dbg_user[DBG_LOCK_USERS];
 			trace_lock_value((lock)->user); \
 			panic(SOF_IPC_PANIC_DEADLOCK); /* lock not acquired */ \
 		} \
-	} while (0);
+	} while (0)
 
 #if DEBUG_LOCKS_VERBOSE
 #define spin_lock_log(lock) \
-	if (lock_dbg_atomic) { \
-		int __i = 0; \
-		int  __count = lock_dbg_atomic >= DBG_LOCK_USERS \
-			? DBG_LOCK_USERS : lock_dbg_atomic; \
-		trace_lock_error("eal"); \
-		trace_lock_value(__LINE__); \
-		trace_lock_value(lock_dbg_atomic); \
-		for (__i = 0; __i < __count; __i++) { \
-			trace_lock_value((lock_dbg_atomic << 24) | \
-				lock_dbg_user[__i]); \
+	do { \
+		if (lock_dbg_atomic) { \
+			int __i = 0; \
```

Comment:
While we are here I think we shouldn't use variables prefixed with `__` since it's for compilers afaik. We can safely use just `i` because it's new scope, but even better to be extra save with different `-Wshadow` flags we can have it with single underscore like `_i` and `_count`.